src/bundle/check.ts

```diff
@@ -0,0 +1,27 @@
+import fs from "fs";
+import path from "path";
+
+const searchInFile = (filePath: string, searchString: string) => {
+  const content = fs.readFileSync(filePath, "utf8");
+  return content.includes(searchString);
+};
+
+export const searchInDirectory = (dirPath: string, searchString: string) => {
+  const files = fs.readdirSync(dirPath);
+  let found = false;
+
+  files.forEach((file) => {
+    const filePath = path.join(dirPath, file);
+    const stats = fs.statSync(filePath);
+
+    if (stats.isDirectory()) {
+      found = searchInDirectory(filePath, searchString);
+    } else if (stats.isFile() && path.extname(filePath) === ".js") {
+      if (searchInFile(filePath, searchString)) {
+        found = true;
```

Comment:
if a subsequent iteration of this loop enters a directory in line 18, it could set `found = false` causing this to fail.

Instead of `files.forEach` I recommend either `files.some()` with an explicit `return boolean` or `for (const file of files) {` with a `return true` when there's a match

Reply:
Something like this

```ts
export const searchInDirectory = (dirPath: string, searchString: string): boolean => {
  const files = fs.readdirSync(dirPath);

  for (const file of files) {
    const filePath = path.join(dirPath, file);
    const stats = fs.statSync(filePath);

    if (stats.isDirectory()) {
      if (searchInDirectory(filePath, searchString)) {
        return true;
      }
    } else if (stats.isFile() && path.extname(filePath) === ".js") {
      if (searchInFile(filePath, searchString)) {
        return true;
      }
    }
  }

  return false;
};
```